presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -134,6 +138,10 @@ public ExchangeClient(
         this.systemMemoryContext = systemMemoryContext;
         this.maxBufferRetainedSizeInBytes = Long.MIN_VALUE;
         this.pageBufferClientCallbackExecutor = requireNonNull(pageBufferClientCallbackExecutor, "pageBufferClientCallbackExecutor is null");
+        this.averageBytesPerRequestEMA = new ExponentialMovingAverage(averageBytesEMAAlpha, DEFAULT_MAX_PAGE_SIZE_IN_BYTES);
+
+        checkArgument(averageBytesEMAAlpha >= 0.0, "averageBytesEMAALpha must be at least 0: %s", averageBytesEMAAlpha);
```

Comment:
Also please move the check before the usage

Reply:
So for some reason unknown to me this is how we do it here https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/ExchangeClientFactory.java#L94

Also this is before "usage" per se because we copy the values but don't use them and the copies are discarded if the check fails.

If we do it differently here compared to ExchangeClientFactory maybe I should add a separate commit moving them first there as well?

Reply:
> If we do it differently here compared to ExchangeClientFactory maybe I should add a separate commit moving them first there as well?

I don't know why is it implemented like that in `ExchangeClientFactory`, but it sounds like a more natural way of doing checks before the usage of the variable being checked. 

It is up to you if you want to fix the existing thing.